File: cpp/326_is_power_of_three/is_power_of_three.cpp
#include <cassert>
#include <tuple>
#include <vector>

class Solution {
 public:
  bool isPowerOfThree(int n) {
    while (n && n % 3 == 0) n /= 3;
    return n == 1;
  }
};

int main() {
  std::vector<std::tuple<int, bool>> tests{
      {27, true},
      {0, false},
      {9, true},
      {45, false},
  };

  for (auto& [n, expected] : tests) {
    assert(Solution().isPowerOfThree(n) == expected);
  }
}

File: rust/326_is_power_of_three/is_power_of_three.rs
struct Solution;

impl Solution {
    pub fn is_power_of_three(mut n: i32) -> bool {
        while n > 0 && n % 3 == 0 {
            n /= 3
        }
        n == 1
    }
}

fn main() {
    let tests = vec![(27, true), (0, false), (9, true), (45, false)];
    for (n, expected) in tests {
        assert_eq!(Solution::is_power_of_three(n), expected);
    }
}
